piksi_tools/interpolate_event_positions.py

```diff
@@ -0,0 +1,246 @@
+  fout= open(outfile,'wt')
+  writer = csv.writer(fout)
+  if msgtype == 'MsgBaselineNED' :
+    indexdata = ("TOW (ms)", "N (mm)", "E (mm)", "D (mm)")
+  elif msgtype == 'MsgPosECEF' or msgtype == 'MsgBaselineECEF' :
+    indexdata = ("TOW (ms)", "X (m)", "Y (m)", "Z (m)")
+  elif msgtype == 'MsgPosLLH' :
+    indexdata = ("TOW (ms)", "Lat (deg)", "Lon (deg)", "Height (m)")
+  writer.writerow(indexdata + ("# of Sats", "Flags"))
+
+  itt=0
+  while itt<numofmsg:
+    if message_type[itt]=="MsgExtEvent" and msg_tow[itt] > 0 :
+      writer.writerow((msg_tow[itt],msg_horizontal[itt],msg_vertical[itt],msg_depth[itt],"0",msg_flag[itt]))
+    itt+=1
+  return 
+
+
+
+
+
+def write_positions(infile, outfile, msgtype, debouncetime):
+  """
+  Organize and output data to log file.
+
+  Parameters
+  ----------
+  infile : string
+    Log file to get data from.
+  outfile : string
+    Output file.
+  msgtype : string
+    type of parameters to analyze and output
+  debouncetime : integer
+    time in milliseconds to compensate for switch debounce 
+  """
+
+  with JSONLogIterator(infile) as log:
+    log = log.next()
+    
+    #declaring all lists 
+    message_type=[]
+    msg_tow=[]
+    msg_horizontal=[]
+    msg_vertical=[]
+    msg_depth=[]
+    msg_sats=[]
+    msg_flag=[]
+    numofmsg=0;
+
+
+
+    while True:
+      try:
+        msg, metadata = log.next()
+        hostdelta = metadata['delta']
+        hosttimestamp = metadata['timestamp']
+        valid_msg=["MsgBaselineECEF","MsgPosECEF","MsgBaselineNED","MsgPosLLH","MsgExtEvent"]
+        #collect all data in lists
+        if msg.__class__.__name__ in valid_msg :
+          message_type.append(msg.__class__.__name__)
+          msg_tow.append(msg.tow)
+          msg_flag.append(msg.flags)
+          if msg.__class__.__name__== "MsgBaselineECEF" or msg.__class__.__name__== "MsgPosECEF" :
+            msg_horizontal.append(msg.x)
+            msg_vertical.append(msg.y)
+            msg_depth.append(msg.z)
+            msg_sats.append(msg.n_sats)
+          elif msg.__class__.__name__== "MsgBaselineNED":
+            msg_horizontal.append(msg.n)
+            msg_vertical.append(msg.e)
+            msg_depth.append(msg.d)
+            msg_sats.append(msg.n_sats)
+          elif msg.__class__.__name__== "MsgPosLLH":
+            msg_horizontal.append(msg.lat)
+            msg_vertical.append(msg.lon)
+            msg_depth.append(msg.height)
+            msg_sats.append(msg.n_sats)
+          elif msg.__class__.__name__ == "MsgExtEvent":
+            print msg.tow
+            msg_horizontal.append("0")
+            msg_vertical.append("0")
+            msg_depth.append("0")
+            msg_sats.append("0")
+          numofmsg+=1
+
+      except StopIteration:
+        print "reached end of file after {0} seconds".format(hostdelta)
+        fix_trigger_rollover(message_type, msg_tow, numofmsg)
+        print 'done roll'
```

Comment:
make these messages more informative to the user.  For example  "processed for rollover error in Piksi event trigger"